lib/aws/log_stack.ts

```diff
@@ -0,0 +1,402 @@
+                    </parse>
+                  </source>`,
+                    "02_filters.conf": "",
+                    "03_dispatch.conf": "",
+                    "04_outputs.conf": `<match hyperswitch.**>
+                  <format>
+                    @type json
+                  </format>
+                  @type copy
+                  <store>
+                    @type stdout
+                  </store>
+                  <store>
+                    @type s3
+                    s3_bucket "#{ENV['S3_BUCKET']}"
+                    s3_region "#{ENV['S3_REGION']}"
+                    path "hyperswitch-logs/%Y/%m/%d/$\{tag\}/"
+                    <buffer tag,time>
+                      @type file
+                      path /var/log/fluent/s3
+                      timekey 3600 # 1 hour partition
+                      timekey_wait 10m
+                      timekey_zone +0530
+                      chunk_limit_size 256m
+                      flush_at_shutdown
+                    </buffer>
+                  </store>
+                </match>`
+
+                },
+            }
+
+        });
+
+        fluentdChart.node.addDependency(sa);
+
+        this.domain = new opensearch.Domain(scope, 'OpenSearch', {
+            version: opensearch.EngineVersion.OPENSEARCH_2_11,
+            enableVersionUpgrade: false,
+            ebs: {
+              volumeSize: 50,
+              volumeType: ec2.EbsDeviceVolumeType.GP3,
+              throughput: 125,
+              iops: 3000,
+            },
+            fineGrainedAccessControl: {
+              masterUserName: "admin",
+              masterUserPassword: cdk.SecretValue.unsafePlainText("Pluentd@123"),
+            },
+            nodeToNodeEncryption: true,
+            encryptionAtRest: {
+              enabled: true,
+            },
+            removalPolicy: cdk.RemovalPolicy.DESTROY,
+            enforceHttps: true,
+            zoneAwareness:{
+              enabled: true,
+              availabilityZoneCount: 2
+            },
+            capacity: {
+              dataNodes: 2,
+              dataNodeInstanceType: "r6g.large.search",
+              multiAzWithStandbyEnabled: false
+            }
+        });
+        // this.domain.grantReadWrite(new iam.AnyPrincipal());
+        const policy = new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          principals: [new iam.AnyPrincipal()],
+          actions: ["es:*"],
+          resources: [`${this.domain.domainArn}/*`],
+        });
+        this.domain.addAccessPolicies(policy);
+
+        const kAnalyticsNS = cluster.addManifest("kube-analytics-ns", {
+            "apiVersion": "v1",
+            "kind": "Namespace",
+            "metadata": {
+                "name": "kube-analytics"
+            }
+        });
+
+        kAnalyticsNS.node.addDependency(this.domain);
+        let open_search_master_user_name = scope.node.tryGetContext('open_search_master_user_name');
```

Comment:
What happens if this isn't passed in the cdk deploy command, Can we handle it please?